API/common/correct.py

```diff
@@ -0,0 +1,187 @@
+            logger.error('Error building Action from {}: {}', desc, e)
+            return None
+
+    @abstractmethod
+    def apply(self, match: MatchResult, command):
+        pass
+
+    @abstractmethod
+    def all_applied(self, command):
+        pass
+
+
+class ReplaceSigAction(Action):
+    def __init__(self, sig: str):
+        self.sig = sig
+
+    def apply(self, match: MatchResult, command):
+        command['command'] = self.sig
+        command['example'] = command['example'].replace(match.sig, self.sig)
+
+    def all_applied(self, command):
+        pass
+
+
+class ReplaceArgAction(Action):
+    def __init__(self, arg_k: str, arg_v: str):
+        self.arg_k = arg_k
+        self.arg_v = arg_v
+
+    def apply(self, match: MatchResult, command):
+        # If corresponding field is `~`, no change in the field
+        if self.arg_k == '~':
+            arg_k = command['arguments'][match.arg_k_idx]
+        else:
+            arg_k = self.arg_k
+        if self.arg_v == '~':
+            # find argument value
+            kv = match.arg_kv.split(' ', 1)
+            arg_v = kv[1] if len(kv) > 1 else ''
+        else:
+            arg_v = self.arg_v
+
+        if arg_k:
+            # Replace the argument
+            command['arguments'][match.arg_k_idx] = arg_k
+            arg_kv = arg_k + ' ' + arg_v if arg_v else arg_k
+            command['example'] = command['example'].replace(match.arg_kv, arg_kv)
+        else:
+            # Remove the argument
+            # We could pop the argument since a traversing is working on the arguments
+            command['arguments'][match.arg_k_idx] = None
+            command['example'].replace(match.arg_kv, '')
+
+    def all_applied(self, command):
+        # Clear the removed arguments
+        command['arguments'] = [arg for arg in command['arguments'] if arg is not None]
+
+
+class AddArgAction(Action):
+    # TODO: The action has problem when arg_k and arg_v is `*` in `MatchRule`
+    def __init__(self, arg_k: str, arg_v: str):
+        self.arg_k = arg_k
+        self.arg_v = arg_v
+
+    def apply(self, match: MatchResult, command):
+        command['arguments'].append(self.arg_k)
+        command['example'] += ' ' + self.arg_k + ' ' + self.arg_v
+
+    def all_applied(self, command):
+        pass
+
+
+class CorrectRule(object):
+    def __init__(self, match_rule: MatchRule, action: Action):
+        self.match_rule = match_rule
+        self.action = action
+
+    def apply(self, command):
+        for match in self.match_rule.match(command):
+            self.action.apply(match, command)
+        self.action.all_applied(command)
+
+
+class CorrectRuleSet(object):
+    def __init__(self, rules: List[CorrectRule]):
+        self.rules = rules
+
+    @staticmethod
+    def load():
+        # An environ example:
+        # "CORRECT_RULE_SET": "[{\"match_rule\": \"('*', '--image', 'UbuntuLTS')\",
+        # \"action\": \"ReplaceArg('~', 'Ubuntu2204')\"}]"
```

Comment:
```suggestion
        ```suggestion
        # * in MatchRule means matching everything.
        # ~ in ReplaceAction means no change on current field.
        # Placeholder information:
        # match_rule(command_signature, parameter, value)
        # ReplaceArg(parameter, value)
        # An environ example:
        # "CORRECT_RULE_SET": "[{\"match_rule\": \"('*', '--image', 'UbuntuLTS')\",
        # \"action\": \"ReplaceArg('~', 'Ubuntu2204')\"}]"
```
```